lib/pliny/librato/metrics/backend.rb

```diff
@@ -24,61 +24,66 @@ def report_measures(measures)
         end
 
         def start
-          start_thread
+          start_counter
+          start_timer
           self
         end
 
         def stop
           metrics_queue.push(POISON_PILL)
-          thread.join
+          timer.terminate
```

Comment:
This is one of many many reasons I don't like working directly with threads and mutexes. They are hard to get right.